src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
```suggestion
const getPersonAge = (person) => person.died - person.born;
const findMother = (child, people) =>  people.find(person => person.name === child.mother);

 const ageDifferanceArray = children.map((child) => getPersonAge(findMother(child, people)) - getPersonAge(child));

  return ageDifferanceArray;
```

Ofc if you prefer you can simplify to `return children.map((child) => getPersonAge(findMother(child, people)) - getPersonAge(child));`